layouts/main_layout.py

```diff
@@ -29,8 +29,33 @@ def gui_layout():
                     }),
                 # URL section
                 html.Div([
-                    dcc.Input(id='url', type='text', value=''),
-                ]),
+                    dcc.Dropdown(
+                        id="url_history_number",
+                        options=[{'label': 10, 'value': 10}, {'label': 20, 'value': 20}, {'label': 50, 'value': 50},
+                                 {'label': 100, 'value': 100}, {'label': 'All', 'value': 999999999999}],
+                        searchable=False,
+                        value=10,
+                        style={
+                            'display': 'table',
+                            'width': '60%',
+                            'height': '25%',
+                            'margin': 'auto',
+                            'backgroundColor': '#FFFFFF'}),
+                    dcc.Dropdown(
+                        id="url_history",
+                        options=[],
+                        searchable=False,
+                        placeholder='Select URL',
+                        style={
+                            'display': 'table',
+                            'width': '60%',
+                            'height': '25%',
+                            'margin': 'auto',
+                            'backgroundColor': '#FFFFFF'}),
+                    dcc.Input(
+                        id="url",
+                        placeholder="Type your URL here",
+                        )]),
```

Comment:
I've checked Input reference page and there's already an implemented feature that allows to display list of possible inputs while allowing user to make custom input.

I've made test script which works pretty nice. I think it should be done with *list* option to url Input:
```
import dash
import dash_html_components as html
import dash_core_components as dcc

external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']

app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
app.layout = html.Div([
    html.Datalist(id="browsers",
                  children=[
                      html.Option(value="chrome"),
                      html.Option(value="firefox"),
                      html.Option(value="internet explorer"),
                  ]),
    dcc.Input(id='demo-dropdown',
              placeholder='Enter a value...',
              type='text',
              value='',
              list='browsers'),
    html.Div(id='dd-output-container')
])


@app.callback(dash.dependencies.Output('dd-output-container', 'children'),
              [dash.dependencies.Input('demo-dropdown', 'value')])
def update_output(value):
    return 'You have selected "{}"'.format(value)


if __name__ == '__main__':
    app.run_server(debug=True)

```